app/models/badges/die_unendliche_geschichte_badge.rb

```diff
@@ -0,0 +1,19 @@
+module Badges
+  class DieUnendlicheGeschichteBadge < Badge
+    seed "Die Unendliche Geschichte",
+      :rare,
+      'die-unendliche-geschichte',
+      'Awarded for submitting 10 iterations to the same exercise'
+
+    def award_to?(user)
+      user.iterations.
+        group(:solution_id).
+        having('COUNT(solution_id) >= 10').
+        exists?
```

Comment:
I've not considered the fact that an iteration could be deleted, as the student _did_ submit an iteration there after all.

Reply:
Yeah, agreed.